nemo/lightning/pytorch/callbacks/model_checkpoint.py

```diff
@@ -251,14 +264,16 @@ def setup(self, trainer, *args, **kwargs) -> None:
         self.async_save = getattr(trainer.strategy, "async_save", False)
         super().setup(trainer, *args, **kwargs)
 
-    def on_save_checkpoint(self, trainer, pl_module, checkpoint):
-        output = super().on_save_checkpoint(trainer, pl_module, checkpoint)
-        return output
-
     def on_train_end(self, trainer, pl_module):
+        from nemo.utils.get_rank import is_global_rank_zero
+
         if trainer.fast_dev_run:
             return None
 
+        if not self.save_optim_on_train_end:
+            ## Do not include optimizer states in final checkpoint
+            trainer.strategy.ckpt_include_optimizer = False
```

Comment:
This flag is not used anywhere?
I think this shouldn't be a modifiable attribute because this could lead to confusion. How about `storage_options` argument to `trainer.strategy.save_checkpoint`?